File: packages/parsers/fixtures/python/behavior.py
"""Fixture for schema-1.2 behavior nodes: raise, exception-handler, assignment, case-branch, explanation."""


def process(value):
    """Process a value, raising on invalid input."""
    result = value * 2
    try:
        if result < 0:
            raise ValueError("negative result")
        return result
    except ValueError as e:
        return -1
    except (TypeError, KeyError) as e:
        return -2


def classify(point):
    """Classify a point via structural matching."""
    match point:
        case Point(x=0, y=0):
            return "origin"
        case Point(x=0, y=_):
            return "on y-axis"
        case _:
            return "other"

File: packages/parsers/fixtures/java/Behavior.java
package com.example.behavior;

/**
 * Behavior demo: exercises schema-1.2 deep-extraction nodes
 * (raise / exception-handler / assignment / case-branch / explanation).
 */
public class Behavior {

    /**
     * Decide on a status code with full fidelity.
     * Throws if negative, maps the rest via a switch.
     */
    public String decide(int status) {
        int count;
        count = status + 1;
        try {
            if (status < 0) {
                throw new IllegalStateException("negative status");
            }
            switch (status) {
                case 0:
                    return "zero";
                case 1:
                    return "one";
                default:
                    return "other";
            }
        } catch (IllegalStateException ex) {
            return "caught";
        }
    }

    public String multi(int x) {
        try {
            if (x == 0) {
                throw new IllegalArgumentException("zero x");
            }
            return "ok";
        } catch (RuntimeException | IllegalStateException e) {
            return "multi";
        }
    }
}